### PR TITLE
Made creator of a repo an admin during repo creation.

### LINKED ISSUE #549: assign repository admin permissions upon save, not in api views
Currently, `assign_user_to_repo_group` is called in the view of the restful API (`rest/views.py`). This means than any other code which creates a repository must individually import and execute `roles.api.assign_user_to_repo_group`. This can be simplified by explicitly calling `roles.api.assign_user_to_repo_group` during `Repository.save()`, either directly or via a signal.

This is currently blocking development of #512. A quick workaround would be to import and execute `roles.api.assign_user_to_repo_group` in the test, but that just adds to the duplication.

The decision to make this change was made by @giocalitri, @noisecapella, and me via Hangout.

### PR DESCRIPTION
**Pull request by ShawnMilo**
No longer a separate step. Allowed for the deletion of code and
simplification of future tests, including for #512.

Closes #549.

FYI: @giocalitri @noisecapella

### CODE REVIEW

File: rest/views.py [hunk 1]
```diff
@@ -95,13 +95,7 @@ def get_queryset(self):
...
     def perform_create(self, serializer):
         """Create a new repository."""
-        # pylint: disable=protected-access
-        repo = serializer.save()
-        assign_user_to_repo_group(
-            self.request.user,
-            repo,
-            GroupTypes.REPO_ADMINISTRATOR
-        )
+        serializer.save()
```
Comment (noisecapella) [inline rest/views.py]: This is the default behavior of `perform_create` so you should be able to just remove this method

File: roles/signals.py [hunk 2]
```diff
@@ -20,3 +20,24 @@ def my_callback(sender, **kwargs):
...
+    """
+    Give the creator of a repository admin permissions
+    over that repository.
+    """
+    from roles.api import assign_user_to_repo_group, roles_init_new_repo
+    from roles.permissions import GroupTypes
+    instance = kwargs.pop("instance", None)
+    if not kwargs["created"]:
+        return
+    if instance.__class__.__name__ != "Repository":
```
Comment (noisecapella) [inline roles/signals.py]: Since we're importing `Repository` anyway can't we compare this against `Repository.__name__` instead?

Comment (noisecapella) [inline roles/signals.py]: Or just compare classes for equality if that works as expected in Python

**Comment by noisecapella** [comment]
Looks good other than the two minor comments

**Comment by ShawnMilo** [comment]
@noisecapella: Thanks. Changes made. I left it as a separate commit for now; I'll flatten when it's good to merge.

**Comment by ShawnMilo** [comment]
I suspect the coverage drop is due to the fact that this PR is mostly deleting code.

**Comment by ShawnMilo** [comment]
Just refreshed from master and flattened. Will merge when Travis passes.

**Pull request merged**